torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1508,7 +1508,7 @@ void ProcessGroupNCCL::watchdogHandler() {
     // the global PG and has globally unique rank ids across trainers.
     dumpPipe.emplace(rank_);
   }
-  while (!done || !terminateProcessGroup_.load()) {
+  while (!done && !terminateProcessGroup_.load()) {
```

Comment:
When terminateProcessGroup_ is true, watchdog will stop working (check workstatus, etc) immediately